scripts/dats_jsonld_annotator/annotator.py

```diff
@@ -0,0 +1,323 @@
+                f"I will overwrite the file now!"
+            )
+        else:
+            logger.warning(
+                f"{dats_jsonld_f.resolve()} already exists and {clobber = }. "
+                f"Consider setting the clobber flag to overwrite existing files"
+            )
+            return
+
+    dats_json = json.load(open(dats_f))
+    schema = json.load(open(schema_f))
+
+    # TODO validate the DATS.json before annotation
+    try:
+        dats_jsonld, context = annotate_dats_object(dats_json, schema, {}, context_dir)
+    except Exception as e:
+        logger.exception(f"Annotating {dats_f} did not complete!", e, exc_info=True)
+
+    # Prefill the context with the SDO mapping
+    context["sdo"] = "https://schema.org/"
+    # Combine the context and the dats graph
+    dats_jsonld["@context"] = [
+        context,
+    ]
+    logger.info(
+        f"Final result written to {dats_jsonld_f.resolve()}! This took {time.time()-tic :.2f} seconds"
+    )
+    json.dump(dats_jsonld, open(dats_jsonld_f, "w"), indent=2)
+
+
+def main(cli_args):
+
+    logging.basicConfig(format="%(levelname)s:  %(message)s", level=logging.INFO)
+    parser = argparse.ArgumentParser(
+        description="Annotate a DATS.json file to DATS.jsonld given a schema"
+    )
+    parser.add_argument(
+        "dats_path",
+        type=pal.Path,
+        help="""
+                If this is a path to a DATS file, then only this DATS file is annotated.
+                If this is a path to a directory, then each subdirectory of this directory is expected
+                to contain a DATS file called DATS.json. Each of these files will then be annotated iteratively
+                """,
+    )
+    parser.add_argument(
+        "-ds",
+        "--dats_schema",
+        type=pal.Path,
+        default=SCHEMA_DIR / "dataset_schema.json",
+    )
+    parser.add_argument("-dc", "--dats_context_dir", type=pal.Path, default=CONTEXT_DIR)
+    parser.add_argument(
+        "--out",
+        type=pal.Path,
+        default=None,
+        help="Where to put the output (default = in the same folder).",
+    )
+    parser.add_argument("--clobber", action="store_true")
+    args = parser.parse_args(cli_args)
+
+    if args.dats_path.is_file():
+        dats_to_jsonld(
+            dats_f=args.dats_path,
+            schema_f=args.dats_schema,
+            context_dir=args.dats_context_dir,
+            out_path=args.out,
+            clobber=args.clobber,
+        )
+    elif args.dats_path.is_dir():
+        files_to_convert = list(args.dats_path.glob("*/DATS.json"))
+        if not args.out.is_dir():
+            logger.warning(f"I will create the output folder at {args.out.resolve()}")
+            args.out.mkdir()
+        if files_to_convert is None:
+            logger.error(
+                f"could not find any DATS.json files in subdirectories of {args.dats_path.resolve()}"
+            )
+        else:
```

Comment:
```suggestion
            )
            sys.exit()
```

Reply:
Instead of adding a long else statement, could exit the script in the if statement above and remove the extra indentation of the else code.